buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -414,6 +414,77 @@ public CompositeByteBuf addComponents(int cIndex, Iterable<ByteBuf> buffers) {
         return addComponents(false, cIndex, buffers);
     }
 
+    /**
+     * Add the given {@link ByteBuf} and increase the {@code writerIndex} if {@code increaseWriterIndex} is
+     * {@code true}. If the provided buffer is a {@link CompositeByteBuf} itself, a "shallow copy" of its
+     * readable components will be performed. Thus the actual number of new components added may vary
+     * and in particular will be zero if the provided buffer is not readable.
+     * <p>
+     * {@link ByteBuf#release()} ownership of {@code buffer} is transferred to this {@link CompositeByteBuf}.
+     * @param buffer the {@link ByteBuf} to add. {@link ByteBuf#release()} ownership is transferred to this
+     * {@link CompositeByteBuf}.
+     */
+    public CompositeByteBuf addFlattenedComponents(boolean increaseWriterIndex, ByteBuf buffer) {
```

Comment:
I'm curious if this can be named just as `addComponents()`.

Reply:
@trustin agree that would be much nicer but unfortunately there are already `addComponents(...)` methods with different semantics - in those cases the components added are 1-1 with the buffer(s) provided, so for example any `CompositeByteBuf` added is just "nested" as-is rather than flattened.

Would welcome other suggestions for a better name though, I don't love `addFlattenedComponents`!

Reply:
unfortunately I dont have any better name atm... @trustin any idea ?

Reply:
Hmm.. not really. Just name it as `addComponents()` and document better? 😅

Reply:
@njhill ^^

Reply:
@normanmaurer sorry, I'd kind of assumed @trustin was joking :). It would be a breaking chance since it would collide with the existing `addComponents(boolean, ByteBuf ...)` method (that performs a "non-flattened" addition). And even with clear doc imo it would still be confusing to have a variety of `addComponents()` methods with different args that all do the same thing and then one which happens to behave differently.

Reply:
true you are right.... lets keep it then.